Bottom: a0c40d507e9bf42ecce88f47f87a44f1e62260fc
Top:    8318a1feda81e1d3a611493bd01b0ea9c1bb3b9b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 20:56:22 +0000

find.c: Fix inverted logic in LookupConnection()

When AndRats is true, we may consider starting with a ratline object,
otherwise we do no. I got this backwards by mistake.


---

diff --git a/src/find.c b/src/find.c
index f58ee63..d8b5a77 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3047,7 +3047,7 @@ LookupConnection (Coord X, Coord Y, bool AndDraw, Coord Range, int which_flag,
   if (type == NO_TYPE)
     {
       type = SearchObjectByLocation (
-        LOOKUP_MORE & ~(AndRats ? RATLINE_TYPE : 0),
+        LOOKUP_MORE & ~(AndRats ? 0 : RATLINE_TYPE),
         &ptr1, &ptr2, &ptr3, X, Y, Range);
       if (type == NO_TYPE)
         return;
